File: examples/elixir/wc.exs
#!/usr/bin/env elixir

# Simple Elixir script to run word count from command line
# Usage: elixir wc.exs <filename>

Code.require_file("lib/word_count.ex")
Code.require_file("lib/word_count/cli.ex")

WordCount.CLI.main(System.argv())

File: examples/ruby/wc.rb
#!/usr/bin/env ruby

require_relative 'lib/word_count'

# Simple command-line interface for the word count program
if ARGV.empty?
  puts "Usage: ruby wc.rb <filename>"
  exit 1
end

filename = ARGV[0]

begin
  wc = WordCount.new(filename)
  puts wc.to_s
rescue Errno::ENOENT
  puts "wc: #{filename}: No such file or directory"
  exit 1
rescue => e
  puts "Error: #{e.message}"
  exit 1
end 